src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5506,42 +5506,43 @@ private static List<CompletionResult> CompleteCommentParameterValue(CompletionCo
         private static readonly HashSet<string> s_dscCollectionVariables =
             new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "SelectedNodes", "AllNodes" };
 
-        internal static List<CompletionResult> CompleteMember(CompletionContext context, bool @static)
+        internal static List<CompletionResult> CompleteMember(CompletionContext context, bool @static, ref int replacementLength)
         {
             // If we get here, we know that either:
-            //   * the cursor appeared immediately after a member access token ('.' or '::').
+            //   * the cursor appeared after a member access token ('.' or '::').
             //   * the parent of the ast on the cursor was a member expression.
             //
             // In the first case, we have 2 possibilities:
             //   * the last ast is an error ast because no member name was entered and we were in expression context
             //   * the last ast is a string constant, with something like:   echo $foo.
 
             var results = new List<CompletionResult>();
-
-            var lastAst = context.RelatedAsts.Last();
-            var lastAstAsMemberExpr = lastAst as MemberExpressionAst;
+            var lastAst = context.RelatedAsts[^1];
```

Comment:
Can this throw if RelatedAsts is empty? Maybe Last() is more safe?

Reply:
Yes it will throw if RelatedAsts is empty, but so will Last().  
Even if Last() returned null instead of throwing there's no scenario where RelatedAsts would be empty because empty strings would throw when TabExpansion2 is called and any character you add will generate at least one RelatedAst.

Reply:
Thanks for clarify. In the case please use Last() as the file style to avoid misunderstanding from code readers.

Reply:
If you insist I'll do it, but my rationale for using the hat operator is that from my own testing it's faster and the coding guidelines explicitly say we should avoid LINQ: https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#performance-considerations  
From my perspective it's just a minor improvement that is no different than some of the recommendations you've made for replacing `(if x == null)` with `(if x is null)` for example.

Reply:
Our top rule is to avoid style and other unrelated changes in functional PRs and keep PRs as small as possible.
For the PR I'd prefer Last() and if you see an performance/style issues please open new PR to replace all the Last().

Sorry for many style requests - it is my duty as maintainer.

Reply:
Ok, I'll update it tomorrow.  
I don't mind the style requests, I like that you try to keep the code clean, and it helps me improve. My confusion stems from an example like this: https://github.com/PowerShell/PowerShell/pull/14550/commits/63989e34fb65703a963ae5c18d8b159d97ed9ea4 where you suggested I update an if statement that I hadn't even written in the first place (I did move it though). To me, that indicated that minor fixes for related code was not only acceptable, but encouraged.

Reply:
> As soon as this API appears, we remove these conditions

Reviewing is human process :-) Every reviewer has personal preferences and sometimes inconsistent :-)
I am always trying to make code clean before area experts start review to save their time.
I understand that this distracts PR authors from functional changes and can frustrate, but after several PRs, authors and reviewers usually work in concert.
Thanks for your patience and understanding.